source/net/net.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package net
+
+import (
+	"net"
+
+	"fmt"
+
+	"io/ioutil"
+	"strings"
+
+	"strconv"
+
+	"github.com/golang/glog"
+)
+
+var physicalInterfacePrefixes = []string{"eth", "em"}
+
+const (
+	speedFeaturePattern = "speed_%d"
+)
+
+// NetSource implements FeatureSource.
+type NetSource struct{}
+
+func (s NetSource) Name() string { return "net" }
+
+// Discover returns feature names for net related features such as network device bandwidth.
+func (s NetSource) Discover() ([]string, error) {
+	features := []string{}
+
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		glog.Errorf("get available network interfaces error. %v", err)
+		return nil, fmt.Errorf("get available network interfaces error")
+	}
+
+	for _, iFace := range interfaces {
```

Comment:
Moving this loop into a function that takes a slice of interfaces as input would allow adding unit test coverage. Could you make this change and add some test cases?

Reply:
Will do.